### PR TITLE
Add support for FreeIPA as a LDAP vendor

### PR DESCRIPTION
**Pull request by mrooding**
## Hey, I just made a Pull Request!

As discussed in #12493 I've added a LDAP vendor for FreeIPA.

I've based the change in `getVendor` on the `getRootDSE` output from FreeIPA:

```text
# extended LDIF
#
# LDAPv3
# base <> (default) with scope baseObject
# filter: (objectclass=*)
# requesting: ALL
#

#
dn:
objectClass: top
namingContexts: dc=intra,dc=organization,dc=io
namingContexts: cn=changelog
namingContexts: o=ipaca
defaultnamingcontext: dc=intra,dc=organization,dc=io
supportedExtension: 2.16.840.1.113730.3.5.7
...ommitted additional supportedExtension...
supportedControl: 2.16.840.1.113730.3.4.2
...ommitted additional supportedControl...
supportedSASLMechanisms: EXTERNAL
supportedSASLMechanisms: GSS-SPNEGO
supportedSASLMechanisms: GSSAPI
supportedSASLMechanisms: DIGEST-MD5
supportedSASLMechanisms: CRAM-MD5
supportedSASLMechanisms: LOGIN
supportedSASLMechanisms: PLAIN
supportedSASLMechanisms: ANONYMOUS
supportedLDAPVersion: 2
supportedLDAPVersion: 3
vendorName: 389 Project
vendorVersion: 389-Directory/1.4.3.231 B2022.061.1818
dataversion: 020220516125028020220516125028020220516125028
netscapemdsuffix: <redacted>
lastusn: 1593209
changeLog: cn=changelog
firstchangenumber: 0
lastchangenumber: 0
ipatopologypluginversion: 1.0
ipatopologyismanaged: on
ipaDomainLevel: 1

# search result
search: 2
result: 0 Success

# numResponses: 2
# numEntries: 1
```

I'm still looking into how to test my change against the actual FreeIPA instance that we have. I couldn't find any documentation on how to use a dev version of Backstage to test specific features. Any pointers would be welcome.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by Rugvip — APPROVED** [review]
Nice, thank you! 👍 

Let's :shipit:

@freben as well for extra 👀 in case something was up

File: docs/integrations/ldap/org.md [hunk 1]
```diff
@@ -12,6 +12,14 @@ groups - directly from an LDAP compatible service. The result is a hierarchy of
 [`Group`](../../features/software-catalog/descriptor-format.md#kind-group) kind
 entities that mirror your org setup.
 
+## Supported vendors
+
+Currently, Backstage only supports the following LDAP vendors:
```
Comment (freben) [inline docs/integrations/ldap/org.md:17]: I'm not sure it's only those. I think it's more like we support pretty much any LDAP vendor that doesn't deviate from some common patterns, right? As in, the FreeIPA system obviously had a super vendor specific ID field offered.

Could we word it more like this?

`Backstage in general supports OpenLDAP compatible vendors, as well as Active Directory and FreeIPA. If you are using a vendor that does not seem to be supported, please [file an issue](https://github.com/backstage/backstage/issues/new?assignees=&labels=enhancement&template=feature_template.md).`

**Review by freben — COMMENTED** [review]
Just a doc related comment. Besides this, looks good!

File: docs/integrations/ldap/org.md (hunk 1 above)
Comment (mrooding) [reply docs/integrations/ldap/org.md]: I was already doubtful about the "default" LDAP implementation. This is a great rewording. Updated the PR

**Comment by mrooding** [comment]
@Rugvip I'm still trying to validate the logic in our environment by using my local version of the plugin. Any pointers on this process? Should a yarn link suffice?

**Comment by Rugvip** [comment]
@mrooding see https://backstage.io/docs/local-dev/linking-local-packages, hopefully there are few enough deps for that not to be a big pain.

**Comment by mrooding** [comment]
Hi @Rugvip, thanks for the link. I should only link `catalog-backend-module-ldap` but no matter what I try it keeps throwing an error when I run `yarn build`:

```
yarn run v1.22.18
$ backstage-cli repo build --all
Building packages

Error: No declaration files found at ../../../developer-portal/backstage/plugins/catalog-backend-module-ldap/src/index.d.ts, be sure to run yarn tsc to generate .d.ts files before packaging
```

My folder structure looks like this:

```
backstage/
developer-portal/
```

Based on the folder structure, I'm adding `"../backstage/plugins/catalog-backend-module-ldap"` to the root package.json and larna.json as stated by the guide. 

What I find peculiar is that it tries to point to the `backstage/plugins/..` folder from within `../../../developer-portal` folder 🤔 What am I doing wrong?

**Pull request merged**